### ISSUE TITLE
When "Hide entry notes by default" is checked "Toggle notes visibility" icon requires two selections before the entry notes field is revealed.

### ISSUE THREAD

**Issue by DavidPesticcio** [keepassxreboot/keepassxc#10126]
## Overview
When "Hide entry notes by default" is checked, in order to enter or edit the notes field the "Toggle notes visibility" icon needs to be selected twice before the entry notes are revealed.

## Steps to Reproduce
1. Ensure "Hide entry notes by default" is checked in Application Settings | Security
2. Create/edit an entry that may or may not have content in the Notes field
3. Select the "Toggle notes visibility" icon to reveal the entry notes - the entry notes are NOT revealed, and the icon does not change
4. Select the "Toggle notes visibility" icon again to reveal the entry notes - the entry notes are revealed, and the icon changes

## Expected Behavior
While creating or editing an entry, with "Hide entry notes by default" checked, selecting the "Toggle notes visibility" icon should reveal the notes field and the icon should change.

## Actual Behavior
While creating or editing an entry, with "Hide entry notes by default" checked, selecting the "Toggle notes visibility" icon requires two selections before the notes field is revealed - the icon only changes on the second selection, but then toggles correctly afterwards while still creating/editing the entry.

## Context
N/A

[NOTE]: # ( Paste debug info from Help → About → Debug Info here )
KeePassXC - Version 2.7.6
Revision: dd21def

Qt 5.15.3
Debugging mode is disabled.

Operating system: Linux Mint 21.1
CPU architecture: x86_64
Kernel: linux 5.15.0-89-generic

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: Linux
Desktop Env: Cinnamon
Windowing System: X11

**Comment by droidmonkey** [comment]
https://github.com/keepassxreboot/keepassxc/assets/2809491/18dcb901-9fe8-49a2-bb6b-9eaeea7fda9e

I cannot replicate this, please post a video showing what is going on, closing in the meantime.

**Comment by 2BeeOrNot** [comment]
V 2.76, macOS 12.7.2, iMac and MBP.

I can replicate this.

Open an entry for editing.
Click the button to show the notes.
Click Cancel or OK.
After that showing the same note or the note of another entry requires two clicks.

If I hide the note and click Cancel or OK showing the same note or another entry note requires one click.

**Comment by droidmonkey** [comment]
Yup that was the key sequence. The notes must be made visible, close the entry view, then reopen.

**Comment by DavidPesticcio** [comment]
I just tried again and I couldn't replicate the behaviour.

After trying several things, it seems to be triggered by modifying and saving a entry note.

I tried changing the title and hitting enter, in the title field, or clicking on OK, and that doesn't seem to be a problem.

Changing the contents of a note, then either clicking on OK, or hitting enter in the title field, does trigger the behaviour.

Locking the database does not resolve the problem, but closing and reopening the database does.
